discretisedfield/plotting/mpl.py

```diff
@@ -0,0 +1,873 @@
+        create plots can be understood with the following code snippet.
+
+        # TODO: Review example after API stabilises.
+
+        .. code-block::
+
+            if ax is None:
+                fig = plt.figure(figsize=figsize)
+                ax = fig.add_subplot(111)
+
+            scalar_field.mpl_scalar(ax=ax, filter_field=scalar_filter_field,
+                                    lightness_field=scalar_lightness_field,
+                                    colorbar=scalar_colorbar,
+                                    colorbar_label=scalar_colorbar_label,
+                                    multiplier=multiplier, cmap=scalar_cmap,
+                                    clim=scalar_clim,)
+
+            vector_field.mpl_vector(ax=ax, use_color=use_vector_color,
+                                    color_field=vector_color_field,
+                                    colorbar=vector_colorbar,
+                                    colorbar_label=vector_colorbar_label,
+                                    multiplier=multiplier, scale=vector_scale,
+                                    cmap=vector_cmap, clim=vector_clim,)
+
+            if filename is not None:
+                plt.savefig(filename, bbox_inches='tight', pad_inches=0.02)
+            ```
+
+            Therefore, to understand the meaning of the arguments which can be
+            passed to this method, please refer to
+            ``discretisedfield.plotting.Mpl.scalar`` and
+            ``discretisedfield.plotting.Mpl.vector`` documentation.
+
+        Example
+        -------
+        .. plot:: :context: close-figs
+
+            1. Visualising the field using ``matplotlib``.
+
+            >>> import discretisedfield as df
+            ...
+            >>> p1 = (0, 0, 0)
+            >>> p2 = (100, 100, 100)
+            >>> n = (10, 10, 10)
+            >>> mesh = df.Mesh(p1=p1, p2=p2, n=n)
+            >>> field = df.Field(mesh, dim=3, value=(1, 2, 0))
+            >>> field.plane(z=50, n=(5, 5)).mpl.plot_simplified()
+
+        .. seealso::
+
+            :py:func:`~discretisedfield.plotting.Mpl.scalar`
+            :py:func:`~discretisedfield.plotting.Mpl.vector`
+
+        """
+        ax = self._setup_axes(ax, figsize)
+
+        multiplier = (uu.si_max_multiplier(self.data.mesh.region.edges)
+                      if multiplier is None else None)
+
+        # Define defaults and update with user-passed dictionaries.
+        scalar_kwargs = {}
+        vector_kwargs = {'use_color': False, 'colorbar': False}
+        scalar_kwargs.update(scalar_args)
+        vector_kwargs.update(vector_args)
+
+        # Set up default scalar and vector fields.
+        # TODO: Do we allow user to specify what scalar and vector fields are? Did we have that before?
+        if self.data.dim == 1:
+            scalar_field = self.data
+            vector_field = None
+
+        elif self.data.dim == 2:
+            scalar_field = self.data
```

Comment:
2D fields are not anymore supported by `scalar`. For 2d fields only `vector` should be used.